src/components/header/renderer.tsx

```diff
@@ -753,13 +773,21 @@ class Header extends React.Component<Props, State> {
                       <div className="button-groups">
                         <button
                           className={this.state.showVega ? 'selected' : ''}
-                          onClick={() => this.setState({ showVega: true })}
+                          onClick={() => {
+                            this.setState({ showVega: true });
+                            const node = ReactDOM.findDOMNode(this.examplePortal.current);
```

Comment:
Can you instead update the redux store here? It seems hacky that you access the Dom.

Reply:
I dint get you exactly.

Are you saying to call the setScrollFunction() here?
@domoritz

Reply:
Just set the redux state. 

Not sure why we need the scroll position in redux tbh.

Reply:
I thought, storing it in redux will be safe, if there is a situation thats makes the component to rerender/reload then the local react state will be lost. 
What do you think?

Reply:
We store the redux state in localstorage. I don't know how often we care about preserving the scroll location. 

Just a thought. I always prefer simple solution.

Reply:
According to me, setting it in the redux store would be fine, because the user will always want to go to the location where he selected the last example.
And if the redux state is stored in the local storage it will make things prettier, because if the user reloads the site, we will have the scroll position and can restore it.

Reply:
> Can you instead update the redux store here? It seems hacky that you access the Dom.

I fear we cannot do that, because the scroll is restored once the portal is opened, and we need to access the DOM to get the scroll back to top.

Reply:
Why not just reset the scoll location  in redux when the user switches from Vega to Vega-Lite and vice versa?

Reply:
We can update the redux store when the user switches.
But the location is fetched from the store only when the portal is opened.
You can see th code, there
the scrollTo is called in onOpen of the react-portal. So updating the redux store here won't make the DOM to scroll up.

As the onOpen function is called only once whenever the portal is opened, and toggling between modes won't make the modal reopen which which not trigger the onOpen function.